### DISCUSSION TITLE
Socket.IO dev proxy

### DISCUSSION THREAD

**Discussion by anthonyma94** [socketio/socket.io#4214] [Q&A]
On my local development, my socket io server is running on port 3000, while my nextjs client is running on port 4200. On production they'll be under one domain. I've tried using next rewrites and http-proxy-middleware to have socket io's traffic proxied to 3000, but with no success. Can anyone instruct me on what to do? Right now I'm doing `process.env.NODE_ENV === "development" ? "http://localhost:3000/" : "/"`, but I'm looking for a cleaner way to achieve that.

**Comment by darrachequesne** [comment]
Hi! Did you find anything? I can't think of a cleaner solution than what you have now.
